include/circt/Dialect/SMT/SMTOps.td

```diff
@@ -222,4 +222,64 @@ def DistinctOp : SMTOp<"distinct", [Pure, SameTypeOperands]> {
   let hasVerifier = true;
 }
 
+def IteOp : SMTOp<"ite", [
+  Pure,
+  AllTypesMatch<["thenValue", "elseValue", "result"]>
+]> {
+  let summary = "a if-then-else function";
+  let description = [{
+    This operation returns its second operand or its third operand depending on
+    whether its first operand is true or not. The semantics are equivalent to
+    the `ite` operator defined in the
+    [Core theory](https://smtlib.cs.uiowa.edu/Theories/Core.smt2) of the SMT-LIB
+    2.6 standard.
+  }];
+
+  let arguments = (ins BoolType:$cond,
+                       AnySMTType:$thenValue,
+                       AnySMTType:$elseValue);
+  let results = (outs AnySMTType:$result);
+
+  let assemblyFormat = [{
+    $cond `,` $thenValue `,` $elseValue attr-dict `:` qualified(type($result))
+  }];
+}
+
+class BoolOp<string mnemonic, string desc> : SMTOp<mnemonic, [Pure]> {
+  let summary = desc;
+  string explanation = [{
+    It supports a variadic number of operands, but requires at least two.
+    This is because the operator is annotated with the `:left-assoc` attribute
+    which means that `op a b c` is equivalent to `(op (op a b) c)`.
+  }];
+  let description = "This operation performs " # desc # [{.
+    The semantics are equivalent to the '}] # mnemonic # [{' operator in the
+    [Core theory](https://smtlib.cs.uiowa.edu/Theories/Core.smt2).
+    of the SMT-LIB Standard 2.6.
+    }] # explanation;
+
+  let arguments = (ins Variadic<BoolType>:$inputs);
+  let results = (outs BoolType:$result);
+
+  let assemblyFormat = "$inputs attr-dict";
+}
+
+def NotOp : BoolOp<"not", "a boolean negation"> {
```

Comment:
To me, this feels weird that this inherit `BoolOp`.
At least, it doesn't feel obvious what aspects of `BoolOp` is actually inherited, since you override the arguments and assemblyFormat.
So I would just inherit from SMTOp directly!